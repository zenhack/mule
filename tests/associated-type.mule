#XFAIL: type member
{ x = {} } : { type t = {}, x : t }
