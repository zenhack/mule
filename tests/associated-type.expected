
Desugared:

	(: (letrec (types ()) (values ((x {}))) (((.= x) {}) x))
	 (record (types ((t (rec t (record (types ()) (values ()))))))
	  (values ((x (rec t (record (types ()) (values ()))))))))

inferred type:

	(record (types ((t (record (types ()) (values ())))))
	 (values ((x (record (types ()) (values ()))))))

Runtime term:

	(LetRec
	 (((0 (Record ())))
	  (App
	   ((App
	     ((Lam
	       (0 () (Lam (1 () (Update (old (Var 1)) (label x) (field (Var 0)))))))
	      (Record ())))
	    (Var 0)))))

Evaluated:

	(Record ((x (Record ()))))
